src/backend/distributed/commands/statistics.c

```diff
@@ -127,6 +132,55 @@ CreateStatisticsStmtObjectAddress(Node *node, bool missingOk)
 }
 
 
+/*
+ * PreprocessDropStatisticsStmt is called during the planning phase for
+ * DROP STATISTICS.
+ */
+List *
+PreprocessDropStatisticsStmt(Node *node, const char *queryString)
+{
+	DropStmt *dropStatisticsStmt = castNode(DropStmt, node);
+	Assert(dropStatisticsStmt->removeType == OBJECT_STATISTIC_EXT);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	QualifyTreeNode((Node *) dropStatisticsStmt);
+
+	Oid relationId;
+	List *ddlJobs = NIL;
+	List *objectNameList = NULL;
+	foreach_ptr(objectNameList, dropStatisticsStmt->objects)
+	{
+		relationId = GetRelIdByStatName(objectNameList);
+
+		if (!OidIsValid(relationId) || !IsCitusTable(relationId))
+		{
+			continue;
+		}
+
+		EnsureCoordinator();
+
```

Comment:
Instead of creating separate ddlJob for every stat object, I suggest creating one job with a drop statistics statement of only distributed statistics;

`DROP STATISTICS dist_stat_1, local_stat, dist_stat_2` -> `DROP STATISTICS dist_stat_1, dist_stat_2`

So here you'll still filter the distributed statistics but then deparse it only once

Reply:
I thought the same at the beginning, but:
`DDLJob` objects have a field named `targetRelationId` which should be set (not a list, just one relationId). And each of these objects in a single `DROP STATISTICS` statement can depend on different relations. That's why I created separate `DDLJob`s for each stat object.

Reply:
what if they're not co-located though? or a reference table?

Reply:
As far as I tested, there's no problem with reference tables. Do you have any specific cases, for reference tables? 

Also I couldn't understand what you mean by being not co-located. Do you mean like one relation is distributed accross the workers and the other one is sharded in the coordinator etc? @marcocitus

Reply:
I probably did not see your comment at the time I wrote mine and was responding to Ozan.

What Ozan is suggesting could be used as an optimization if 2 distributed tables are co-located, but in general we need to split into separate DDL jobs to support reference tables, non-co-located tables.